### ISSUE TITLE
Mammoth Harness (New Card)

### ISSUE THREAD

**Issue by kelsioboy** [magarena/magarena#1444]
name=Mammoth Harness
image=https://magiccards.info/scans/en/hl/64.jpg
value=1.299
rarity=R
type=Enchantment
subtype=Aura
cost={3}{G}
ability=Enchant creature;
timing=aura
enchant=default,neg creature
requires_groovy_code=Grounded;Mammoth_Harness
oracle=Enchant creature\nEnchanted creature loses flying.\nWhenever enchanted creature blocks or becomes blocked by a creature, the other creature gains first strike until end of turn.

[
    new BlocksTrigger() {
        @Override
        public MagicEvent executeTrigger(final MagicGame game,final MagicPermanent permanent,final MagicPermanent blocker) {
            final MagicPermanent enchantedCreature = permanent.getEnchantedPermanent();
            final MagicPermanent blocked = blocker.getBlockedCreature();
            final MagicPermanent target = enchantedCreature == blocker ? blocker.getBlockedCreature() : blocker;
            final boolean isInvolved = enchantedCreature == blocker || enchantedCreature == blocked
            return (isInvolved && target.isValid()) ?
                new MagicEvent(
                    permanent,
                    target,
                    this,
                    "RN gains first strike until end of turn."
                ):
                MagicEvent.NONE;
        }
        @Override
        public void executeEvent(final MagicGame game, final MagicEvent event) {
            event.processRefPermanent(game, {
                game.doAction(new GainAbilityAction(it, MagicAbility.FirstStrike));
            });
        }
    }
]
